Balance updates

STOPTHECOUNT opened the pull request:
Added correct tooltips. Some of these are not actually used for the tooltips in-game but it can't hurt to have the correct values there.

STOPTHECOUNT committed:
Change end-game boss

Damage: 9999 -> 300
Attack rate: 4 -> 0.4
Added splash radius: 100
Added splash factor: 0.2

STOPTHECOUNT committed:
Buff Gargoyle

Closes #93

STOPTHECOUNT committed:
Buff Whelp

Closes #92

STOPTHECOUNT committed:
Buff Ice Troll

Closes #94

STOPTHECOUNT committed:
Buff Warlock

Closes #95

STOPTHECOUNT committed:
Buff Komodo

Closes #86

STOPTHECOUNT committed:
Buff Trident

Closes #87

STOPTHECOUNT committed:
Buff Krogoth

Closes #85

STOPTHECOUNT committed:
Buff Cyborg

Closes #84

STOPTHECOUNT committed:
Buff Greymane

Closes #83

STOPTHECOUNT committed:
Buff Grizzly

Closes #82

STOPTHECOUNT committed:
Buff Thunderbird

Closes #81

STOPTHECOUNT committed:
Buff Golem

Closes #80

STOPTHECOUNT committed:
Buff Mudman

Closes #78

STOPTHECOUNT committed:
Buff Frenzy Ghoul

Closes #77

STOPTHECOUNT committed:
Buff Archer

Closes #75

STOPTHECOUNT committed:
Buff Geomancer

Closes #74

STOPTHECOUNT committed:
Buff Frostwolf

Closes #72

STOPTHECOUNT committed:
Buff Peasant

Closes #69

STOPTHECOUNT committed:
Buff Fangtooth

Closes #67

STOPTHECOUNT committed:
Buff Hatchling

Closes #65

STOPTHECOUNT committed:
Buff Swamprunner

Closes #66

STOPTHECOUNT committed:
Nerf Gnoll Warrior

Closes #64

STOPTHECOUNT committed:
Buff Destroyer

Closes #37

STOPTHECOUNT committed:
Buff Cyborg/Krogoth

Closes #23

STOPTHECOUNT committed:
Buff Medicine Man

Closes #22

STOPTHECOUNT committed:
Buff Swordmage

Closes #17

STOPTHECOUNT committed:
Komodo Tooltip

STOPTHECOUNT committed:
Trident Tooltip

STOPTHECOUNT committed:
Krogoth Tooltip

STOPTHECOUNT committed:
Cyborg Tooltip

STOPTHECOUNT committed:
Greymane Tooltip

STOPTHECOUNT committed:
Grizzly Tooltip

STOPTHECOUNT committed:
Thunderbird Tooltip

STOPTHECOUNT committed:
Golem Tooltip

STOPTHECOUNT committed:
Mudman Tooltip

STOPTHECOUNT committed:
Fenzy Ghoul Tooltip

STOPTHECOUNT committed:
Archer Tooltip

STOPTHECOUNT committed:
Geomancer Tooltip

STOPTHECOUNT committed:
Frostwolf Tooltip

STOPTHECOUNT committed:
Peasant Tooltip

STOPTHECOUNT committed:
Fangtooth Tooltip

STOPTHECOUNT committed:
Hatchling Tooltip

STOPTHECOUNT committed:
Swamprunner Tooltip

STOPTHECOUNT committed:
Gnoll Warrior Tooltip

STOPTHECOUNT committed:
Gnoll Warrior Typo

STOPTHECOUNT committed:
Destroyer Tooltip

STOPTHECOUNT committed:
Cyborg Tooltip

STOPTHECOUNT committed:
Medicine Man Tooltip

STOPTHECOUNT committed:
Swordmage Tooltip

STOPTHECOUNT committed:
Remove 'Economic Growth'-Button from 'War Room'

Economic Growth is not a game mode any more.

STOPTHECOUNT committed:
Fix map description "Author:....OZE Team" to "....Team OZE"

Resolves #103

STOPTHECOUNT committed:
Update King

Max. Attack: 30
Max. HP: 50
Max. Regen.: 30
Autoheal will now always heal to 100% HP.

STOPTHECOUNT committed:
Update Game Modes

Resolves #98 and #101
Also updates the whole gamemode-processing, texts visible to user at start, increases start counter from 20s to 30s.
Commented out many parts that are no longer used instead of deleting them, so that later code-cleaning will be easier.

marcoabreu commented:
I think you accidentally have multiple changes in the same PR. Would you mind rebasing so that we have the different changes in the different PRs?

STOPTHECOUNT commented:
Yes

STOPTHECOUNT committed:
Improving text for champion leak

Champion: 'Don't you dare use anarchy on me!'

STOPTHECOUNT committed:
Updating text for final round

Was: "This is Final Boss !!"
Now: "This is the final round!"

STOPTHECOUNT committed:
Buff Lvl 11

Closes #89

STOPTHECOUNT committed:
Nerf Lvl 4

Closes #88

STOPTHECOUNT committed:
Nerf Lvl 14

Closes #90

STOPTHECOUNT committed:
Buff Lvl 15

Closes #91

STOPTHECOUNT committed:
Buff Hades

Closes #15

STOPTHECOUNT committed:
Buff Elite Archer

Closes #19
Attackspeed: +10%
HP: +100

STOPTHECOUNT committed:
Buff Violet

#18 Buffed HP, but not Gust Dmg.

STOPTHECOUNT committed:
Buff Ogre Pulverizer

Closes #70

STOPTHECOUNT committed:
Buff Nightcrawler

Closes #71

STOPTHECOUNT committed:
Buff Pandaren

Closes #73

STOPTHECOUNT committed:
Correcting Gold selling values for previous commits

STOPTHECOUNT committed:
Reduce Hero Cost

Closes #63
Maybe this needs to be adjusted in the future but I like the fact that heroes are back!

STOPTHECOUNT committed:
Mirror king ability

Closes #3

STOPTHECOUNT committed:
Mirror king ability

Closes #3

STOPTHECOUNT committed:
Fix Lvl 3 Champ Bug

Closes #97
Tricky to find. Champions get additional HP by eating tomes. The tome they eat depends on the HP of the unit OR the round. Probably HP for lvl 3 was increased from <145 to 200 at some point.
I also made it less prone the errors for future unit changes.

STOPTHECOUNT committed:
Buff Violet Gust Dmg

Closes #18

STOPTHECOUNT committed:
Buff Ghost Part 1

#14 spell range back to 400

STOPTHECOUNT committed:
Update Quest Info

All quests texts are now in one place. Updated texts, renamed variables and functions)
Text still needs to be improved so that beginners better understand the game.

STOPTHECOUNT committed:
Reduce Champion Reflect Damage (Spike/Thorns)

Closes #79

STOPTHECOUNT committed:
King HP on Leaderboard only for own King

Closes #112

STOPTHECOUNT committed:
Rewrote the whole ReRoll-function and some related stuff

Was horrible, now it's bad.

STOPTHECOUNT committed:
Revert unwanted change in previous commit

STOPTHECOUNT committed:
Introducing testversion-flag

Development-versions now have additional options:
-ap (so you can easier test all units)
-07 (so you can directly test each level [e.g. lvl 7])
If you start at a higher level, building time is increased and you have unlimited amount of gold and lumber.

STOPTHECOUNT committed:
Remove Food Cost from Dark Ranger

STOPTHECOUNT committed:
Remove "x20" from game name shown if no upkeep

STOPTHECOUNT committed:
Fix bug in Reroll-function

STOPTHECOUNT committed:
Rewrote lots of texts (content, when they appear, duration)

Goal is to avoid spamming in the beginning and have more useful information for beginners.
Still needs to be improved.

STOPTHECOUNT committed:
Hatchling Tooltip

STOPTHECOUNT committed:
Fixes Telescope on Wanderer

Closes #113

STOPTHECOUNT committed:
Testversion message every round

STOPTHECOUNT committed:
Typo in Tooltip

STOPTHECOUNT commented:
> 
> 
> Could we print a message every round when testVersion is true? Just to be sure that it doesn't get missed

Good idea, just implemented that.

STOPTHECOUNT committed:
Always enable -start command for test version

STOPTHECOUNT committed:
Slightly increasing readability by adding new line after each function

STOPTHECOUNT committed:
Slightly increasing readability by using proper indentation

STOPTHECOUNT committed:
Update README.md

STOPTHECOUNT committed:
More readability improvements

Variable renamed
Corrected indentation for single-line-block
New line before else and elseif

STOPTHECOUNT committed:
More readability by correcting some indentation issues

STOPTHECOUNT committed:
Renaming variable 'Unit'

STOPTHECOUNT committed:
Some code cleaning

STOPTHECOUNT committed:
Deleting lots of stuff related to economic growth

STOPTHECOUNT committed:
Deleting lots of stuff related to limited income

STOPTHECOUNT committed:
Deleting lots of stuff related to host pick mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -ar mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -sd mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -gg mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -mm and -ah modes

STOPTHECOUNT committed:
Deleting lots of stuff related to -hg mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -eq mode

STOPTHECOUNT committed:
Deleting lots of stuff related to -nm mode

STOPTHECOUNT committed:
Deleting lots of stuff related to unused modes

STOPTHECOUNT committed:
Deleting lots of stuff related to -nc mode

STOPTHECOUNT committed:
Mirror mode only allowed with -pr

Closes #116
Other code improvements around game modes

STOPTHECOUNT committed:
Change tooltip command-attack and king's taunt

Closes #118

STOPTHECOUNT committed:
Remove food cost soul of hero

Closes #119

STOPTHECOUNT committed:
Removing variable and improving multiboard

STOPTHECOUNT committed:
Deleted about 9,000 lines of code related to cheats or useless stuff

I'm sure there is still a lot of useless stuff in there.

STOPTHECOUNT committed:
Deleted more useless code

STOPTHECOUNT committed:
Bringing quick -cls back

STOPTHECOUNT committed:
Fix Kings spells

STOPTHECOUNT committed:
Update map info and version

STOPTHECOUNT committed:
Fix roll visibility

Closes #125

STOPTHECOUNT committed:
Fixes lots of tooltips, still a lot to do

Also closes #129

STOPTHECOUNT committed:
Fix Destroyer tooltip

Closes #130

STOPTHECOUNT committed:
Fix Paladin hero cost and remove food cost

Closes #122
Also fixes 2 other hero costs

STOPTHECOUNT committed:
Remove immolation from Doom

Closes #131

STOPTHECOUNT committed:
Buff Lvl 33

Closes #124

STOPTHECOUNT committed:
Buff Gargoyle and Whelp

Closes #121

STOPTHECOUNT committed:
Update map description

Closes #104

STOPTHECOUNT committed:
Remove internal cooldown for all spellcasters

Closes #99

STOPTHECOUNT committed:
Buffed/nerfed a few units

Closes #120

STOPTHECOUNT committed:
Fixes problem that roll refresh can spawn same unit

Closes #7

STOPTHECOUNT committed:
Renaming a couple of variables and minor code improvements

STOPTHECOUNT committed:
Fixes problem that roll refresh can spawn same unit 2

This should close #7 now

STOPTHECOUNT committed:
Deleting useless stuff and renaming lots of variables

STOPTHECOUNT committed:
Deleteted some more useless variables and functions

STOPTHECOUNT committed:
Fixes Gnoll Tooltip and Hatchling HP

STOPTHECOUNT committed:
Fixes cooldown for kingspells

STOPTHECOUNT committed:
No champs without -cc -ac

Also some more stuff deleted

STOPTHECOUNT committed:
Fixes cooldown for Command Attack and King's Taunt

STOPTHECOUNT committed:
Fixes late unit teleport for team west

Closes #117

STOPTHECOUNT committed:
Fixes tooltip for Polar Bear and Magnataur

STOPTHECOUNT committed:
Renaming lots of variables and functions, also deleting some stuff

Some variables need to be renamed/reorganized later, but I think it's good first step towards more readability.

STOPTHECOUNT committed:
Change anarchy cooldown to 10 sec

STOPTHECOUNT committed:
Adding player colors for 13-24

Might be useful at some point

STOPTHECOUNT committed:
Renaming some variables

STOPTHECOUNT committed:
Removing some redundancies

STOPTHECOUNT committed:
Fixing two binaries that caused the missing observers

#100
Now 14 Obvservers are possible

STOPTHECOUNT committed:
Change standard zoom and zoom formula

Player default now at 100% and range is 50% -> 150% (100% is ~151 with old formula)
Observer default now at 120% (not changable for observers)

STOPTHECOUNT committed:
Fixes cooldown for Violet (Blue Wind)

STOPTHECOUNT committed:
Fixed cooldown for withdraw ability of units

STOPTHECOUNT committed:
Improved some tooltips

STOPTHECOUNT committed:
Renaming Kings

STOPTHECOUNT committed:
Renaming some functions

STOPTHECOUNT committed:
Fixes cooldown for Komodo and Trident submerge

STOPTHECOUNT committed:
Renaming some functions

STOPTHECOUNT committed:
Remove War Room unless it's -cc mode

Also renaming it

STOPTHECOUNT committed:
Removing stuff related to Player(12)

It is probably useless and might cause problems with an observer in that spot

STOPTHECOUNT committed:
Removing stuff related to Player(15)

It is probably useless and might cause problems with an observer in that spot

STOPTHECOUNT committed:
Minor code cleaning

STOPTHECOUNT committed:
Rearrange mode-processing

Now if you type -cc it will be -prcc, because -pr is the default.
Also the messages appear in the same order as on the multiboard.

STOPTHECOUNT committed:
Removed bug

STOPTHECOUNT committed:
Buff Ghost Part 2

It's 1% every 2 seconds now with 400 range

STOPTHECOUNT committed:
Fixes Infernal HP from LoD

STOPTHECOUNT committed:
Improves some tooltips

STOPTHECOUNT committed:
Fixes Ghost tooltip and adds range

STOPTHECOUNT committed:
Fixes cooldown for Dark and Royal Presence

STOPTHECOUNT committed:
Fixes Goblin Blaster cooldown

STOPTHECOUNT committed:
Buff Ghost range to 550

STOPTHECOUNT committed:
Fixes units not showing their damage during building time

They are now invulnerable

STOPTHECOUNT committed:
Improve some messages and tooltips

STOPTHECOUNT committed:
Improving hundreds of tooltips

STOPTHECOUNT committed:
Fixes ally unit attack

STOPTHECOUNT committed:
Fixes missing income

STOPTHECOUNT committed:
Sets command attack dooldown to 10 seconds

STOPTHECOUNT committed:
Fixes bug in game mode processing

STOPTHECOUNT committed:
Removes Altar of Heroes for now

Planning to bring it back in 6.1

STOPTHECOUNT committed:
Fixes cooldown of Guardian Spirit

This also fixes the bug that u sometimes suddenly had Guardian Spirit on all units in Level 35

STOPTHECOUNT committed:
Prepares some general things for Hero Dread Lord

STOPTHECOUNT committed:
Message and tooltips improved

STOPTHECOUNT committed:
Removes mana bar from units that don't use mana

- Ascendant
- Sword Mage
- Tuskarr Spearman
- Ice Troll
- Ice Troll Shaman

STOPTHECOUNT committed:
All mana units show full mana instantly after building and during fighting

STOPTHECOUNT committed:
Allow repick in -ap Test-Mode

STOPTHECOUNT committed:
Fixes creep HP for all levels

STOPTHECOUNT committed:
Fixes Sphere and Storm Geyser cooldown for Highborn

STOPTHECOUNT committed:
King Info function rewritten

- Max level
- Better colors
- Anarchy info
- Presence info
- Better layout
Also some messages related to king info are improved

STOPTHECOUNT committed:
Sword Mage back to 200g

Closes #135

STOPTHECOUNT committed:
Reduces Guardian Entangle by 2 seconds on boss

STOPTHECOUNT committed:
Deleted redundant or useless files

STOPTHECOUNT committed:
Improves some texts

Also closes #138

STOPTHECOUNT committed:
Deletes empty files

STOPTHECOUNT committed:
Updates loading screen

STOPTHECOUNT committed:
Minor changes

STOPTHECOUNT committed:
Fixes desync problems

STOPTHECOUNT committed:
Fixes King HP not shown properly for observers

STOPTHECOUNT committed:
Simplifications

STOPTHECOUNT committed:
Buff Veteran

Closes #142

STOPTHECOUNT committed:
Buff Ghost

Closes #139

STOPTHECOUNT committed:
Updates minimap

STOPTHECOUNT committed:
Reduces text size of game modes

STOPTHECOUNT committed:
Fixes visible terrain mode

Prior game settings are now taken into account

STOPTHECOUNT committed:
Buff some levels

Level 9: +50 HP
Level 18: +150 HP
Level 19: +100 HP and +4 damage

STOPTHECOUNT committed:
Improves Warden tooltip and fixes Defiler mana regen

STOPTHECOUNT committed:
Removes King's Autocast-ability

STOPTHECOUNT committed:
Fixes King control getting stuck to leaver

Closes #5
If "king-owner" leaves then control of the king returns to himself

STOPTHECOUNT committed:
Improves tooltip and message

STOPTHECOUNT committed:
Adds file that wrongly got deleted

STOPTHECOUNT committed:
Removes "4.5" behind "LEGION" on terrain

STOPTHECOUNT committed:
Improves text renames variable

STOPTHECOUNT committed:
Fixes desync on -ap test mode

STOPTHECOUNT committed:
Removes mana bar from Tree of Battle

STOPTHECOUNT committed:
Improves all unit tooltips and binds HP to unit's stats

STOPTHECOUNT committed:
Fixes format in UnitWeapons.slk

STOPTHECOUNT committed:
Fixes all inconsistencies between Fighter, Builder and Tooltip

STOPTHECOUNT committed:
Updates some Tooltips

Closes #146
Rest was done before by auto-replacement

STOPTHECOUNT committed:
Buff Defiler Range

Closes #147
Also improves some tooltips

STOPTHECOUNT committed:
Defiler Tooltip

STOPTHECOUNT committed:
Fixes inconsistencies with Attack and Defense Type
